src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -745,18 +760,30 @@ private CachedCase createNewCase(Long uprn) throws CTPException {
     // Query AIMS for UPRN
     AddressIndexAddressCompositeDTO address = addressSvc.uprnQuery(uprn);
 
-    if (address.getCountryCode() == SCOTTISH_COUNTRY_CODE) {
+    if (SCOTTISH_COUNTRY_CODE.equals(address.getCountryCode())) {
       log.with("uprn", uprn)
           .with("countryCode", address.getCountryCode())
           .warn("Scottish address retrieved");
       throw new CTPException(Fault.VALIDATION_FAILED, "Scottish address found for UPRN: " + uprn);
     }
 
+    CachedCase cachedCase = caseDTOMapper.map(address, CachedCase.class);
+
+    try {
+      cachedCase.setCaseType(CaseType.valueOf(address.getCensusAddressType()));
+    } catch (IllegalArgumentException e) {
+      log.with("uprn", uprn)
+          .with("AddressType", address.getCensusAddressType())
+          .warn("AIMs AddressType not valid");
+      throw new CTPException(Fault.RESOURCE_NOT_FOUND, e, "AIMs AddressType not valid: " + uprn);
```

Comment:
Could we make this "AddressType of 'Floating Caravan Palace' not valid for Census"

More descriptive and Serco won't know what AIMS is

Reply:
Done.